spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/BaseBatchReader.java

```diff
@@ -125,4 +129,28 @@ private CloseableIterable<ColumnarBatch> newOrcIterable(
         .withNameMapping(nameMapping())
         .build();
   }
+
+  private Schema vectorizationSchema(SparkDeleteFilter deleteFilter) {
+    // For pos delete, deleteFilter has appended _pos to the required schema.
+    // For example, SELECT id, data FROM test, the requested schema is id and data. If there
+    // is position delete, deleteFilter will append _pos to the schema so the schema becomes
+    // id, data and _pos. However, vectorization reader only needs to read the requested columns,
+    // i.e. id and data, so we want to remove the _pos from the schema when building the
+    // vectorization reader. Before removing _pos, we need to make sure _pos is not explicitly
+    // selected in the query.
+    if (deleteFilter != null) {
+      if (deleteFilter.hasPosDeletes() && expectedSchema().findType("_pos") == null) {
```

Comment:
Do we have a const for `_pos`?

Reply:
Yes, we do. Changed to `MetadataColumns.ROW_POSITION.name()`